backend/src/controllers/ratings.controller.ts

```diff
@@ -0,0 +1,64 @@
+import { NextFunction, Request, Response } from 'express';
+import { CreateRatingsDto } from '@dtos/ratings.dto';
+import { Ratings } from '@interfaces/ratings.interface';
+import ratingService from '@services/ratings.service';
+
+class RatingController {
+  public ratingService = new ratingService();
+
+  public getRatingsByCustomerId = async (req: Request, res: Response, next: NextFunction) => {
+    try {
+      const customerId: string = req.params.id;
+      const findRatings: Ratings[] = await this.ratingService.getRatingsByCustomerId(customerId);
+
+      res.status(200).json({ data: findRatings, message: 'findAll' });
+    } catch (error) {
+      next(error);
+    }
+  };
+
+  public getRatingsByProductId = async (req: Request, res: Response, next: NextFunction) => {
+    try {
+      const productId: string = req.params.id;
+      const findRatings: Ratings[] = await this.ratingService.getRatingsByProductId(productId);
+
+      let ratings = 0;
+      length = findRatings.length;
+      for (let i = 0; i < length; i = i + 1) {
```

Comment:
Works, but is not the best way in js.
Javascript has many array functions that can help you. 
I think a reduce approach like this should be a "better" solution.

Even there is an error with ratingsSum, 0 will be retuned as default AVG.

only as a small improvement

````javascript
let ratingsSum = findRatings.reduce(
  (previousValue, currentValue) => previousValue + currentValue.rating,
  0
);

const ratingAVG = ratingsSum / findRatings.length || 0;

const data = [ratingAVG, findRatings.length];
```

Reply:
[3aeae49] done